app/indexers/hyrax/file_set_indexer.rb

```diff
@@ -30,6 +30,7 @@ def generate_solr_document # rubocop:disable Metrics/AbcSize, Metrics/MethodLeng
         solr_doc['original_checksum_tesim'] = object.original_checksum
         solr_doc['alpha_channels_ssi']      = object.alpha_channels
         solr_doc['original_file_id_ssi']    = original_file_id
+        solr_doc['generic_type_si'] = 'FileSet'
```

Comment:
The `generic_type` field is used in `PcdmMemberPresenterFactory`.  This is called from:
* `Hyrax::WorkShowPresenter #member_presenter_factory` which only calls it for `Valkyrie::Resource`
* `Hyrax::WorkFormHelper #form_file_set_select_for(parent:)`  which is called from several views.
  * select representative_id in _form_representative.html.erb
  * select rendering_ids in _form_rendering.html.erb
  * select thumbnail_id in _form_thumbnail.html.erb

Reply:
I tested with Main and with this PR's branch.  The behavior for GenericWork is the same for both branches.  GenericWork uses a different member_presenter_factory that does not depend on `generic_type_si`, so the addition of this index field has no impact on non-valkyrie works.